### ISSUE TITLE
SHA256 hash of the TOS has changed

### ISSUE THREAD

**Issue by thomastweets** [kuba/simp_le#114]
With the update to the Let’s Encrypt Subscriber Agreement today (v.1.1.1) the SHA256 hash has changed to 6373439b9f29d67a5cd4d18cbc7f264809342dbf21cb2ba2fc7588df987a6221
I can create a pull request to change line 894/895 of simp_le.py, however, I am not sure whether this covers all of it (and I am not deep into the topic).

**Comment by LeoFidjeland** [comment]
Having this issue as well, cannot create any new certs at all.

Please approve the PR asap.

And thanks for all the great work :)

**Comment by andresmrm** [comment]
Also got same error here.

@LeoFidjeland : As an workaround you can pass parameter:
`--tos_sha256 6373439b9f29d67a5cd4d18cbc7f264809342dbf21cb2ba2fc7588df987a6221`.
At least here it worked.

**Comment by madshargreave** [comment]
+1, my app is down because I cannot renew my certificate. Any help is greatly appreciated

**Comment by amiryal** [comment]
To the best of my understanding, this issue only affects new registrations, and it can be worked around by using `--tos_sha256 6373439...`, as suggested by @andresmrm above. If using an existing (already registered) account key, then no changes are necessary. Correct?

**Comment by thomastweets** [comment]
@amiryal True! However, some other software that makes use of simp_le does not allow to specify this parameter. (e.g. https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion or https://github.com/dokku/dokku-letsencrypt). I am working on a forked docker image of https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion to have a work-around for now as I indeed need to do new registrations with simp_le.

**Comment by thomastweets** [comment]
Here we go: https://hub.docker.com/r/thomastweets/docker-letsencrypt-nginx-proxy-companion/

**Comment by LeoFidjeland** [comment]
Awesome, thanks so much @thomastweets! Works perfectly!

You saved the day!!

**Comment by vrozkovec** [comment]
Is this project still maintained?

**Comment by chicks-net** [comment]
@vrozkovec -- the author has been inactive on github since June.  Maybe google has been keeping him busy.

**Comment by Thor77** [comment]
Does anyone have contact to the maintainer?
If not, does anyone plan to create a organization for this project and start maintaining it again?

**Comment by chicks-net** [comment]
I gave up and I've migrated to [acme.sh](https://github.com/Neilpang/acme.sh).

**Comment by NikLP** [comment]
Tried acme based on above suggestion but it's so poorly documented I couldn't get anywhere and am coming back to this as my domain is down! If someone wants to expand on the the one-line POS examples littered randomly about the web for acme then please do so and let me know!! I'm running Apache/Debian and happy to help with any efforts on that platform.

**Comment by wiz78** [comment]
If you want to workaround the problem, you can comment a couple of lines in **simp_le/venv/lib/python/site-packages/acme/messages.py**:

```
        #if not set(canon_jobj).issubset(self._REGISTERED_TYPES): # <-- this line (should be 140)
            # TODO: acme-spec is not clear about this: 'It is a JSON
            # dictionary, whose keys are the "resource" values listed
            # in {{https-requests}}'z
         #   raise ValueError('Wrong directory fields') # <-- and this one
```

**Comment by amiryal** [comment]
@wiz78 the right way to do it is to upgrade the `acme` library to a later version, as implemented by [this fork of simp_le](https://github.com/zenhack/simp_le). If you would like, you may suggest your change to [ACME upstream](https://github.com/certbot/certbot/tree/master/acme).

**Comment by wiz78** [comment]
@amiryal you're right. Mine was meant to be a dirty hack, since I was under the impression that the whole project had been abandoned.

**Comment by amiryal** [comment]
Glad to help inform people about the more maintained fork. :)
